.devcontainer/README.md

```diff
@@ -0,0 +1,217 @@
+
+# CCCL Dev Containers
+
+To ensure consistency and ease of setup, the CUDA C++ Libraries (CCCL) team uses [Development Containers](https://code.visualstudio.com/docs/devcontainers/containers).
+This guide offers information on setting up container in Visual Studio Code and manually with Docker.
+
+## Quickstart: VSCode (Recommended)
+
+### **Prerequisites**:
+- [Visual Studio Code](https://code.visualstudio.com/)
+- [Remote - Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)
+
+### **Step 1: Clone the Repository**
+
+- **Easy Way**:
+Copy and paste the following URI into your browser to automatically clone and open in VSCode:
+
+    ```
+    vscode://vscode.git/clone?url=https://github.com/nvidia/cccl.git
+    ```
+
+- **Manual Way**:
+  ```bash
+  git clone https://github.com/nvidia/cccl.git
+  ```
+
+### **Step 2: Launch Development Environment in VSCode**
+
+- Open the cloned directory in VSCode.
+- A prompt will appear in the lower right corner suggesting to "Reopen in Container". Click it.
+- A list of devcontainers will be shown. These are named with a pattern `cuda<CTK_VERSION>-<HOST-COMPILER>`. Select the desired environment to start coding!
+
+## Quickstart: Docker (Manual Approach)
+
+### **Prerequisites**:
+- [Docker](https://docs.docker.com/desktop/install/linux-install/)
+
+
+### **Step 1: Clone the Repository**
+```bash
+git clone https://github.com/nvidia/cccl.git
+```
+
+### **Step 2: Launch Docker Container**
+Navigate to the cloned directory and use the [`launch.sh`](./launch.sh) script:
+
+```bash
+cd cccl
+./.devcontainer/launch.sh --docker
+```
+
+This script initiates a Docker container tailored for the specified CUDA and compiler versions. Inside, you'll get an interactive shell session with the `cccl` repository already mounted, allowing direct editing and testing.
+
+For a deeper dive into dev containers, configurations, and benefits, continue reading below.
+
+## Overview
+
+The CUDA C++ Libraries (CCCL) team uses [Development Containers, or "Dev Containers"](https://code.visualstudio.com/docs/devcontainers/containers), to provide a consistent, reproducible, and isolated development environment shared for both local development and CI.
+We encourage external contributors to use them as they provide an effortless onramp to setting up your environment for contributions. With just a few actions in Visual Studio Code or a basic Docker command, you're ready in the same environment the CCCL team works in.
+
+If Docker feels daunting, Dev Containers make it more intuitive. They manage Docker's intricacies, freeing you to code. No need for complex Docker commands or profound containerization knowledge. The `devcontainer.json` file dictates each environment, designating a base Docker image with the desired CUDA Toolkit (CTK) and host compiler version. This file might also detail extra configurations, tools, or settings for further refinement.
```

Comment:
> They manage Docker's intricacies, freeing you to code. No need for complex Docker commands or profound containerization knowledge.

I'd revise this to sound a little less magical. It's setting sky-high expectations that will feel like a major letdown if something goes awry.